aptos-move/aptos-vm/src/move_vm_ext/session/view_with_change_set.rs

```diff
@@ -231,6 +231,62 @@ impl<'r> TResourceView for ExecutorViewWithChangeSet<'r> {
                 .get_resource_state_value_metadata(state_key),
         }
     }
+
+    fn get_resource_state_value_size(&self, state_key: &Self::Key) -> PartialVMResult<u64> {
+        match self.change_set.resource_write_set().get(state_key) {
+            Some(
+                AbstractResourceWriteOp::Write(write_op)
+                | AbstractResourceWriteOp::WriteWithDelayedFields(WriteWithDelayedFieldsOp {
+                    write_op,
+                    ..
+                }),
+            ) => Ok(write_op.bytes_size() as u64),
+            // We could either return from the read, or do the base read again.
+            Some(AbstractResourceWriteOp::InPlaceDelayedFieldChange(_)) | None => self
+                .base_executor_view
+                .get_resource_state_value_size(state_key),
+            Some(AbstractResourceWriteOp::WriteResourceGroup(_))
+            | Some(AbstractResourceWriteOp::ResourceGroupInPlaceDelayedFieldChange(_)) => {
+                // In case this is a resource group, and feature is enabled that creates these ops,
+                // this should never be called.
+                // Call to metadata should go through get_resource_state_value_metadata(), and
+                // calls to individual tagged resources should go through their trait.
+                Err(code_invariant_error(format!(
+                    "get_resource_state_value_size called for resource group with key {:?}",
+                    state_key
+                ))
+                .into())
+            },
+        }
+    }
+
+    fn resource_exists(&self, state_key: &Self::Key) -> PartialVMResult<bool> {
+        match self.change_set.resource_write_set().get(state_key) {
```

Comment:
there's a lot of duplicate code here, maybe there should be a function to return write_op or forward_to_base_view

Reply:
absolutely, the boilerplate actually got a lot better. Thanks!